wrangler-service/src/main/java/co/cask/wrangler/service/kafka/KafkaService.java

```diff
@@ -322,14 +425,86 @@ public void specification(HttpServiceRequest request, final HttpServiceResponder
 
       JsonArray values = new JsonArray();
       values.add(value);
-      response.addProperty("status", HttpURLConnection.HTTP_OK);
-      response.addProperty("message", "Success");
-      response.addProperty("count", values.size());
-      response.add("values", values);
+      response.addProperty(PropertyIds.STATUS, HttpURLConnection.HTTP_OK);
+      response.addProperty(PropertyIds.MESSAGE, PropertyIds.SUCCESS);
+      response.addProperty(PropertyIds.COUNT, values.size());
+      response.add(PropertyIds.VALUES, values);
       sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
-    } catch (Exception e) {
+    } catch (Throwable e) {
+      LOG.error("Exception while setting plugin specification", e);
       error(responder, e.getMessage());
     }
   }
   
+  class FetchData<K, V> {
+
+	  KafkaConsumer<K, V> consumer;
+
+	  FetchData(KafkaConsumer<K, V> consumer) {
+		  this.consumer = consumer;
+	  }
+
+	  public List<Row> fetchData(String topic, Integer count) throws Exception {
+		  List<Row> recs = new ArrayList<>();
+		  consumer.subscribe(Lists.newArrayList(topic));
+		  try {
+			  ConsumerRecords<K, V> records = consumer.poll(10000);
+			  LOG.debug("Record count received is: {}", records.count());
+			  for (ConsumerRecord<K, V> record : records) {
+				  Row rec = new Row();
+				  V value = record.value();
+				  if (value instanceof GenericRecord) {
+					  parseAvroRecord(rec, value);
+				  } else {
+					  rec.add(outputFields.get(0), value);
+				  }
+
+				  recs.add(rec);
+				  if (count < 0) {
+					  break;
+				  }
+				  count--;
+			  }
+		  } catch (Exception e) {
+			  LOG.error("Exception while reading data from topic {}, {}", topic, e.getMessage());
+			  throw e;
+		  } finally {
+			  consumer.close();
+		  }
+		  return recs;
+	  }
+
+	  private void parseAvroRecord(Row rec, V value) {
+		  GenericRecord messageRecord = (GenericRecord) value;
+		  List<Field> fields = messageRecord.getSchema().getFields();
+		  for (Field field : fields) {
+			  String fieldName = field.name();
+
+			  if (outputFields.contains(fieldName)) {
+				  Object fieldValue = messageRecord.get(fieldName);
+				  // In case of String values it was observed that sometimes
+				  // value from kafka
+				  // comes as type Utf8 which is not a serializable class.
+				  // Convert this to string in the
+				  // builder
+				  if (fieldValue instanceof org.apache.avro.util.Utf8) {
+					  fieldValue = ((org.apache.avro.util.Utf8) fieldValue).toString();
+				  } else if (fieldValue instanceof GenericData.Array) {
+					  parseArray(fieldValue);
+				  }
+				  rec.add(fieldName, fieldValue);
+			  }
+		  }
```

Comment:
What if some of the fields present in 'outputFields' are not present in 'fields' object? Is it okay to just return only those fields which are present in both 'fields' as well 'outputFields' ?

Reply:
https://guavus-jira.atlassian.net/browse/RAFD-3488